src/osp/core/global_id.h

```diff
@@ -24,26 +24,31 @@
  */
 #pragma once
 
-#include <cstddef>
+#include <longeron/utility/enum_traits.hpp> // for lgrn::underlying_int_type_t
+
+#include <type_traits> // for std::is_integral_v
 
 namespace osp
 {
 
 template <typename ID_T, typename DUMMY_T = void>
 struct GlobalIdReg
 {
-    [[nodiscard]] static inline ID_T create() noexcept
+    using underlying_t = lgrn::underlying_int_type_t<ID_T>;
+    static_assert(std::is_integral_v<underlying_t>);
+
+    [[nodiscard]] static constexpr ID_T create() noexcept
     {
-        return ID_T(sm_count++);
+        return ID_T{sm_count++};
     }
 
-    [[nodiscard]] static inline std::size_t size() noexcept
+    [[nodiscard]] static constexpr underlying_t size() noexcept
```

Comment:
I want this to return std::size_t since this is always used to resize containers

Reply:
doing that causes warnings in almost every place that the size() function is used.

implicit promotion to size_t almost never warns, but implicit narrowing always does.

Reply:
well i guess i'm not seeing what the problem is if the thing being returned implicitly converts to std::size_t?

Here it's used to allow looping over all the instances of the id.


https://github.com/TheOpenSpaceProgram/osp-magnum/blob/a747cab17ae3f5bea02a800c8b9cfbe8a1e345c3/src/testapp/sessions/vehicles.cpp#L286-L289

same here

https://github.com/TheOpenSpaceProgram/osp-magnum/blob/a747cab17ae3f5bea02a800c8b9cfbe8a1e345c3/src/testapp/sessions/vehicles.cpp#L445-L455

Perhaps we need to add a new `largest()` function or something that returns the largest value as the type?

Reply:
I see. These don't yet use the strong ID types. Ideally I'd prefer something like
```cpp
for (TypeId const id : TypeReg_t::ids()) {...}
```

though I'd agree to just leave this function as underlying_t for now.

Reply:
I did add a `largest()` function, but you can rename it if you want in a followup PR, no need to get a review.